src/context/cdlist.cpp

```diff
@@ -0,0 +1,55 @@
+/*********************                                                        */
+/*! \file cdlist.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andres Noetzli
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Context-dependent list class (only supports append)
+ **
+ ** This source file implements the specializations for CDList<bool> because
+ ** the underlying std::vector<bool> is a specialization of std::vector<T> that
+ ** is not strictly compatible.
+ **
+ ** See https://en.cppreference.com/w/cpp/container/vector_bool for details
+ ** about std::vector<bool>.
+ **/
+
+#include "context/cdlist.h"
+
+namespace CVC4 {
+namespace context {
+
+template <>
+void CDList<bool, DefaultCleanUp<bool> >::truncateList(const size_t size)
```

Comment:
Are there any CDList<bool>'s existing already? If not we could simply ban that expansion for now  with a static_assert and a comment.